packages/suite/src/hooks/suite/useWindowVisibility.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+import { updateWindowVisibility } from 'src/actions/suite/windowActions';
+
+import { useDispatch } from './useDispatch';
+
+export const useWindowVisibility = () => {
+    const dispatch = useDispatch();
+
+    const onWindowVisibilityChange = () => {
```

Comment:
it would be possible to extend this by also having state from electron layer, there is one more state for macOS, that it also detects when app is cover by another app, but that would make it more complicated for now. Let's tested at least using just Page visibility api from renderer layer